### PR TITLE
Initial Linked Form Step Support

### PR DESCRIPTION
**Pull request by r4vi**
This is still a WIP but it'd be good to have you give initial impressions

### CODE REVIEW

File: src/server/plugins/register-form/steps/date-of-birth.js [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+const fields = [
+  {id: 'day', label: 'Day'},
+  {id: 'month', label: 'Month'},
+  {id: 'year', label: 'Year'}
+];
+
+const schema = Joi.object().keys({
+  'day': Joi.number().integer().min(1).max(31).required(),
+  'month': Joi.number().integer().min(1).max(12).required(),
+  'year': Joi.number().integer().min(1885).max(2025).required(),
```
Comment (r4vi) [inline src/server/plugins/register-form/steps/date-of-birth.js]: need to make .max( `the year of today's date` )

File: src/server/plugins/register-form/steps/date-of-birth.js [hunk 2]
```diff
@@ -0,0 +1,33 @@
+import Joi from 'joi';
+import {postHandlerFactory, getHandlerFactory} from './common';
+
+const fields = [
```
Comment (r4vi) [inline src/server/plugins/register-form/steps/date-of-birth.js]: could probably merge `fields` and `schema` into a single object

File: src/server/plugins/register-form/steps/date-of-birth.js [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+  'year': Joi.number().integer().min(1885).max(2025).required(),
+  'submit': Joi.any().optional().strip()
+});
+
+const title = 'What is your date of birth?';
+const key = 'dateOfBirth';
+
+const handlers = {
+  GET: getHandlerFactory(key, fields, title, schema),
+  POST: postHandlerFactory(key, fields, title, schema, 'dateOfBirth')
```
Comment (r4vi) [inline src/server/plugins/register-form/steps/date-of-birth.js]: want to have this `nextStep` parameter provided from the order of the `steps` definition

File: src/server/plugins/register-form/steps/date-of-birth.js [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+};
+
+exports.options = {
+  title,
+  fields,
```
Comment (marcofucci) [inline src/server/plugins/register-form/steps/date-of-birth.js]: I don't think you need to export `fields` and `schema`.

Comment (r4vi) [inline src/server/plugins/register-form/steps/date-of-birth.js]: I want to merge them both

File: src/server/plugins/register-form/steps/date-of-birth.js (hunk 3 above)
Comment (r4vi) [inline src/server/plugins/register-form/steps/date-of-birth.js]: yeah I was thinking the same

File: src/server/plugins/register-form/steps/date-of-birth.js (hunk 4 above)
Comment (marcofucci) [inline src/server/plugins/register-form/steps/date-of-birth.js]: I mean, you don't need to export them even if merged. 
They are only useful to the `*HandlerFactory` functions, not externally to this module.

**Comment by r4vi** [comment]
yeah I guess I'm being lazy and exporting them because I want to import
into tests; but I can do a conditional export if NODE_ENV=test

**Comment by r4vi** [comment]
TODO:
 
- [ ] update README with SESSION_SECRET env var
- [ ] allow yarn run dev to start without having a webpackassets.json
- [ ] when running in dev mode, show errors in response instead of 'internal server error'

**Pull request merged**